### DISCUSSION TITLE
No ringing back when calling

### DISCUSSION THREAD

**Discussion by wujingke** [IchthysMaranatha/asterisk-chan-quectel#51] [Q&A]
<img width="1195" alt="image" src="https://github.com/IchthysMaranatha/asterisk-chan-quectel/assets/565778/b48450c5-d227-4b73-b518-898920271461">

When I made an outbound call, there was no reply.

**Comment by IchthysMaranatha** [comment]
First make sure modem is working using minicom without asterisk, you should be able to place calls and answer them. Then follow [this](https://github.com/IchthysMaranatha/asterisk-chan-quectel/discussions/13)

**Reply by wujingke** [reply]
I can make a call, but there is no ringing sound after dialing. There is a sound when the call is answered, and there is a sound when the call is rejected. There is no ringing back after dialing.

**Comment by IchthysMaranatha** [comment]
Which modem? Have you followed the test bed setup?

**Comment by IchthysMaranatha** [comment]
Sorry, saw your pic now

**Comment by IchthysMaranatha** [comment]
Try uac mode also

**Comment by IchthysMaranatha** [comment]
And troubleshoot using only one modem

**Comment by wujingke** [comment]
Activate UAC but still no ringing sound   
<img width="910" alt="image" src="https://github.com/IchthysMaranatha/asterisk-chan-quectel/assets/565778/6e0d2fed-c577-48cd-b58e-deda8cdec7cc">
<img width="472" alt="image" src="https://github.com/IchthysMaranatha/asterisk-chan-quectel/assets/565778/79f523a2-8607-4022-ae71-500ee95ea30b">
<img width="679" alt="image" src="https://github.com/IchthysMaranatha/asterisk-chan-quectel/assets/565778/ae99dcb6-210a-4877-9e0d-7ed25d549d37">

**Comment by IchthysMaranatha** [comment]
If you have something like pulseaudio that will prevent asterisk from taking control. 

First run testbed setup on new fresh ubuntu install according to steps given without any additional software and identify source of issue.

**Reply by wujingke** [reply]
There is just no ringing back. There is sound when connected, which is very strange.

**Reply by wujingke** [reply]
The system is newly built

**Comment by IchthysMaranatha** [comment]
All done according to test bed with nothing extra?

**Comment by IchthysMaranatha** [comment]
Please follow this without any deviation https://github.com/IchthysMaranatha/asterisk-chan-quectel/discussions/13

**Reply by wujingke** [reply]
According to the documentation, no other additions

**Comment by IchthysMaranatha** [comment]
What do asterisk logs say?

**Comment by IchthysMaranatha** [comment]
What does quectel show device quectel0 say?

**Reply by wujingke** [reply]
<img width="1048" alt="image" src="https://github.com/IchthysMaranatha/asterisk-chan-quectel/assets/565778/a9215601-18b4-4fa5-ae4a-fc1c521cc59e">

**Reply by wujingke** [reply]
<img width="1807" alt="image" src="https://github.com/IchthysMaranatha/asterisk-chan-quectel/assets/565778/1236f16e-a74b-44a2-9ed6-8e9d30d2fdd3">

**Comment by wujingke** [comment]
Found the problem

Add in asterisk's sip.conf
directmedia=no
prematuremedia=yes
progressinband=never

**Comment by IchthysMaranatha** [comment]
Had you done this from guide?

In settings make sure network is set to bridged adapter

This ensures that all devices are in same network connected to your router without asterisk having to proxy. If it needs to proxy, directmedia=no

**Comment by wujingke** [comment]
After testing, it was found that when the called party has activated a ring tone, the operator returns 183 and sends rtp. When the called party does not have a ring tone, the operator sends 180 and does not send rtp, so there is no ring tone back. Is there any good solution for this? The module will not return 180 or 183

**Comment by IchthysMaranatha** [comment]
You can force asterisk to generate ringback with r option in Dial

**Reply by wujingke** [reply]
After forcibly setting the ringback tone, the machine is shut down and the prompt tones such as empty numbers cannot be heard.

**Comment by IchthysMaranatha** [comment]
When building asterisk, have you included all sounds needed with relevant codec?

**Comment by IchthysMaranatha** [comment]
Also you might need to set up indications.conf
